python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -141,24 +160,50 @@ def fetch_object(self, obj_id: str):
 
         Args:
             obj_id: The object ID of the GPU object.
+            use_object_store: Whether to use the object store to fetch the GPU object.
 
         Returns:
             None
         """
         from ray.experimental.gpu_object_manager.gpu_object_store import (
             __ray_fetch_gpu_object__,
         )
+        from ray.experimental.collective import get_tensor_transport_manager
 
         if self.gpu_object_store.has_object(obj_id):
             return
-
         gpu_object_meta = self.managed_gpu_object_metadata[obj_id]
         src_actor = gpu_object_meta.src_actor
-        tensors = ray.get(
-            src_actor.__ray_call__.options(concurrency_group="_ray_system").remote(
-                __ray_fetch_gpu_object__, obj_id
-            )
+        tensor_transport_backend = gpu_object_meta.tensor_transport_backend
+        tensor_transport_manager = get_tensor_transport_manager(
+            tensor_transport_backend
         )
+        if use_object_store or not tensor_transport_manager.is_one_sided():
```

Comment:
The behavior we want is to always use the tensor transport that was defined at @ray.method time, unless it's overridden at ray.get time. I don't think this would do that for "nccl" transport?